### DISCUSSION TITLE
Is there any integrated extention in vscode or any tool in vscode could help do the static code analysis on prisma

### DISCUSSION THREAD

**Discussion by jiz17043** [prisma/prisma#24230] [Q&A] [answered]
### Question

Is there any integrated extention in vscode or any tool in vscode could help do the static code analysis on prisma

**Comment by ludralph** [comment, marked as answer]
Hi @jiz17043 👋 

Thank you for raising this question.

There is a Prisma extension for Visual Studio Code that can assist with static code analysis. The Prisma VS Code extension uses the prisma-fmt engine, which is written in Rust, to parse and format the Prisma schema, perform static analysis (linting), and provide feedback such as errors and suggestions.

The extension also provides support for users when writing or editing their Prisma files, offering features such as autocompletion, syntax highlighting, linting, and documentation. It also helps with navigating and understanding the schema by providing tooltips, comments, and the ability to go to definitions.

You can install the official [Prisma VS Code extension](https://marketplace.visualstudio.com/items?itemName=Prisma.prisma) from the VS Code marketplace.

**Reply by jiz17043** [reply]
Thanks for your reply.
